tests/test_finite_diff.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+import pytest
+
+import pennylane as qml
+
+
+x1 = 0.5
+y1 = -0.3
+
+x2 = np.array([0.5, -0.1975])
+y2 = -0.3
+
+
+@pytest.mark.parametrize(
+    ("x", "y", "argnum", "idx", "delta", "exp_grad"),
+    [
+        (x1, y1, 0, None, 0.01, -47.341084643123565),
+        (x1, y1, 1, None, 0.02, -371.7453193214464),
+        # (x2, y2, 0, [0], 0.01, np.array([np.array([-47.34108464, 0.0]), 0])),
+    ],
+)
+def test_first_finit_diff(x, y, argnum, idx, delta, exp_grad):
+    r"""Tests the correctness of the first-order finite difference function."""
+
+    def f(x, y):
+        return np.sin(x) / x ** 4 + y ** -3
+
+    grad = qml.finite_diff(f, argnum=argnum, idx=idx, delta=delta)(x, y)
+
+    assert np.allclose(np.array(grad, dtype=float), exp_grad)
+
+
+x1 = 1.975
+y1 = 0.33
+
+x2 = np.array([1.975, 0.33])
```

Comment:
Do we want a test case with multi dimensional x? since the indexing in the finite diff functions now seem to be general enough to take multi dim args, eg. `x3 = np.array([[1,2], [3,4]])` and `idx = [[0,1], [1,1]]`

Reply:
Agree with Zy

Reply:
Thank you @zeyueN, test added.